sim/src/mechanics/intersection.rs

```diff
@@ -561,6 +570,14 @@ impl IntersectionSimState {
                 }
                 (None, None) => {}
             }
+
+            // It's unlikely, but the player might create/destroy traffic signals close together and
+            // change the uber-turns that exist. To be safe, recalculate everywhere.
+            state.uber_turn_neighbors.clear();
+            if let Some(mut set) = map_model::IntersectionCluster::autodetect(state.id, map) {
```

Comment:
I'm not very familiar with this code... but it seems like `state` is per intersection, right?

So if we edit *this* intersection, we invalidate `self.state.uber_turn_neighbors`. That makes sense to me.

But if editing this intersection changed the uber turns which exist, couldn't this intersection have been newly added to/removed from nearby uber turns - in which case the *other* intersections in that affected uber turn need to invalidate *their* `uber_turn_neighbors` as well? 

Or am I missing something?

Reply:
> in which case the other intersections in that affected uber turn need to invalidate their uber_turn_neighbors as well

Correct, we potentially need to recalculate `uber_turn_neighbors` for all intersections. That's what this loop does. We iterate over all `for state in self.state.values_mut()` and then totally recalc from scratch.

The calculation does `map_model::IntersectionCluster::autodetect`, which doesn't know about `uber_turn_neighbors` at all; it's not like we have to propagate the change out in a certain order or anything. Detecting clusters of intersections is only based on the map, not any sim state

Reply:
Ah, thank you for resolving my confusion. `self` here  is the "per map" `intersection::IntersectionSimState`. `self.state` is **every** per-intersection `state`.